### PR TITLE
Fixed CCNode _checkListeners & CCScheduler update

### PR DESCRIPTION
**Pull request by yoki0805**
Changes:
 * CCNode _checkListeners function. The loop index should reset to zero.
 * CCScheduler update function.  When first time, it should call update function once to avoid skip  current frame.

### CODE REVIEW

File: cocos2d/core/CCScheduler.js [hunk 1]
```diff
@@ -200,6 +200,7 @@ proto.update = function (dt) {
     if (this._elapsed === -1) {
         this._elapsed = 0;
         this._timesExecuted = 0;
+        this.update(dt);
```
Comment (pandamicro) [inline cocos2d/core/CCScheduler.js]: It is the designed behavior, the scheduled tasks will always start after the first time interval ends.

**Review by pandamicro — CHANGES_REQUESTED** [review]
I think the modification to _checkListener is necessary, but the original code seems intended, I want to ask in which scenario we can see problems causing by the current implementation

File: cocos2d/core/CCScheduler.js (hunk 1 above)
Comment (yoki0805) [reply cocos2d/core/CCScheduler.js]: > It is the designed behavior, the scheduled tasks will always start after the first time interval ends.

I coded this, and the task ran after 2 frames. So, the  behavior is correct?

**Comment by yoki0805** [comment]
> I think the modification to _checkListener is necessary, but the original code seems intended, I want to ask in which scenario we can see problems causing by the current implementation

No, just for safety and readability.
When the node._bubblingListeners are check failed, the result of checking node._capturingListeners could be wrong. Or insure that the node._bubblingListeners and the node._capturingListeners are mutex.

File: cocos2d/core/CCScheduler.js (hunk 1 above)
Comment (pandamicro) [reply cocos2d/core/CCScheduler.js]: Yes, actually the second "Component updated" and the first "myFunc updated" is printed in the same frame. The scheduler execution is later than the component update. So the behavior in each frame is like this:

- Frame 0:
    - start -> myFunc scheduled
    - update
- Frame 1:
    - update
    - myFunc call
- repeat Frame 1

**Comment by pandamicro** [comment]
> When the node._bubblingListeners are check failed, the result of checking node._capturingListeners could be wrong. Or insure that the node._bubblingListeners and the node._capturingListeners are mutex.

I agree, all events should be checked in both capturing listener and bubbling listener

File: cocos2d/core/CCScheduler.js (hunk 1 above)
Comment (yoki0805) [reply cocos2d/core/CCScheduler.js]: Okay, I got it.
I misunderstood  the log. Lol.
Thanks for your patience. :)

Comment (pandamicro) [reply cocos2d/core/CCScheduler.js]: So please revert this line, and we can merge your pr

Comment (yoki0805) [reply cocos2d/core/CCScheduler.js]: I got the 403 problem. Now it reverted.

**Review by pandamicro — APPROVED** [review]

**Review by caryliu1999 — APPROVED** [review]

**Comment by SantyWang** [comment]
Thanks for your contribution. Please make another pr to v2.4.0-release. By the way, do you have an account of cocos forum, if so, you can leave your account id and we will give you a label

**Comment by pandamicro** [comment]
@holycanvas I rebased the PR to v2.4.0, you can do it too with the "Edit" button at the right of the PR title

Oh, I just notice you have done it, why you change it back ?

**Comment by SantyWang** [comment]
> Oh, I just notice you have done it, why you change it back ?

There will be some useless commits

**Comment by yoki0805** [comment]
Here you are.
[https://github.com/cocos-creator/engine/pull/6905](url)

**Pull request closed without merging**